IPython/core/displayhook.py

```diff
@@ -168,6 +174,10 @@ def write_format_data(self, format_dict, md_dict=None):
         md_dict : dict (optional)
             The metadata dict to be associated with the display data.
         """
+        image_data = format_dict.get("image/png")
+        if image_data is not None and self.view_image(image_data):
+            return
+
```

Comment:
Hi @Carreau,

I moved the code from core/displayhook.py to the RichPromptDisplayHook in terminal/prompts.py . Hope this is better.